### ISSUE TITLE
Error when `pip install tendenci` called `subprocess-exited-with-error`

### ISSUE THREAD

**Issue by moaz-mokhtar** [tendenci/tendenci#1130]
Hello all,

When following instructions to install tendenci by `pip install tendenci` found below error when installing `gevent` library. I did research about that error but I couldn't found a clear answer.

I did follow instruction in <https://tendenci.readthedocs.io/en/latest/installation/installation.html#section-2-development-setup> and <https://github.com/tendenci/deploy_tendenci>.

---
 
**Device info**
Host Machine: Ubuntu 22.04.1 LTS
Guest Machine ( Virtual - QEMU ): Ubuntu 22.04.1 LTS

---

**Error**
```shell
Collecting gevent==21.1.2
  Using cached gevent-21.1.2.tar.gz (5.9 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [60 lines of output]
      warning: src/gevent/resolver/cares.pyx:38:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/resolver/cares.pyx:40:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/resolver/cares.pyx:41:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      Compiling src/gevent/resolver/cares.pyx because it changed.
      [1/1] Cythonizing src/gevent/resolver/cares.pyx
      warning: src/gevent/libev/corecext.pyx:326:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:809:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:811:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:813:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:817:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      Compiling src/gevent/libev/corecext.pyx because it changed.
      [1/1] Cythonizing src/gevent/libev/corecext.pyx
      Compiling src/gevent/_greenlet_primitives.py because it changed.
      [1/1] Cythonizing src/gevent/_greenlet_primitives.py
      Compiling src/gevent/_hub_primitives.py because it changed.
      [1/1] Cythonizing src/gevent/_hub_primitives.py
      Compiling src/gevent/_hub_local.py because it changed.
      [1/1] Cythonizing src/gevent/_hub_local.py
      Compiling src/gevent/_waiter.py because it changed.
      [1/1] Cythonizing src/gevent/_waiter.py
      warning: src/gevent/_gevent_cgreenlet.pxd:112:33: Declarations should not be declared inline.
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      cdef load_traceback
      cdef Waiter
      cdef wait
      cdef iwait
      cdef reraise
      cpdef GEVENT_CONFIG
            ^
      ------------------------------------------------------------
      
      src/gevent/_gevent_cgreenlet.pxd:182:6: Variables cannot be declared with 'cpdef'. Use 'cdef' instead.
      Compiling src/gevent/greenlet.py because it changed.
      [1/1] Cythonizing src/gevent/greenlet.py
      Traceback (most recent call last):
        File "/srv/venv_mysite/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/srv/venv_mysite/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/srv/venv_mysite/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-xsdqge85/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-xsdqge85/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-xsdqge85/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 482, in run_setup
          super(_BuildMetaLegacyBackend,
        File "/tmp/pip-build-env-xsdqge85/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 335, in run_setup
          exec(code, locals())
        File "<string>", line 294, in <module>
        File "/tmp/pip-install-heegk0m7/gevent_40f90278bb3a4c5ca06d3e8d803225eb/_setuputils.py", line 237, in cythonize1
          new_ext = cythonize(
        File "/tmp/pip-build-env-xsdqge85/overlay/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1129, in cythonize
          cythonize_one(*args)
        File "/tmp/pip-build-env-xsdqge85/overlay/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1296, in cythonize_one
          raise CompileError(None, pyx_file)
      Cython.Compiler.Errors.CompileError: src/gevent/greenlet.py
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.


```

**Comment by jennyq** [comment]
@moaz-mokhtar,

Thank you for reporting the issue! Currently we haven't tested it on Ubuntu 22.04 yet. Will look into it.

**Comment by moaz-mokhtar** [comment]
Issue resolved but found another issues and resolved by installing specific libraries.

--- 

**Briefly**
- Install Python 3.8
- Create virtual environment with python 3.8
- Install library `python3.8-distutils` will resolve `apt install tendenci` issue
- Another library to be installed `python3.8-dev` to fix `psycopg2` issue.

---

**Comands**
```shell
# Install python 3.8
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt install python3.8

# Create virtual environment for python 3.8
virtualenv --python="/usr/bin/python3.8" "/path/to/venv/"

...

sudo apt install -y python3.8-distutils

sudo aptinstall -y python3.8-dev

...

```

**Comment by jennyq** [comment]
Awesome, you've figured it out!

**Comment by jennyq** [comment]
Update: This issue has been resolved in the latest Tendenci version 14.3.2. No need to use python 3.8 as a workaround anymore.
